tools/automate_gcp_os_version.py

```diff
@@ -0,0 +1,118 @@
+"""
+checks for the latest os and updates the os versions in vars.tf file in dc-only, 
+multi-region and single connector deployments.
+Use this command to run this script in cloud shell: python3 tools/update_gcp_os.py
+"""
+import argparse
+import datetime
+import os
+import re
+import subprocess
+from subprocess import CalledProcessError
+
+COMPUTE_SERVICES_ENABLE = "gcloud services enable compute.googleapis.com"
+COMMANDS = ["gcloud compute images list"]
+PATHS = [["single-connector", "vars.tf"],
+         ["multi-region", "vars.tf"],
+         ["dc-only", "vars.tf"]]
+
+# commands = ["gcloud compute images list | grep windows-server-2016",
+#             'gcloud compute images list | grep centos-7',
+#             'gcloud compute images list | grep ubuntu-1804-bionic']
+
+
+# Added env variable for future purpose
+def get_os(env='cloud'):
+    """
+    get the latest version from GCP
+    """
+    try:
+        recent_os = []
+        if env == "cloud":
+            # breakpoint()
+            for command in COMMANDS:
+                capture_output = subprocess.run(command.split(' '), check=True,
+                                                capture_output=True).stdout.decode().split('\n')
+                recent_os.extend([key[0] for key in map(lambda x: x.split(' '), capture_output)])
+        return recent_os
+    except (AttributeError, TypeError, CalledProcessError) as ex:
+        print("Exception occurred: ", ex)
+
+
+def compare_versions(recent_os, line):
+    """
+    Checks for version numbers and returns true if latest version available
+    """
+    try:
+        # version_date variable is the current os version in the vars.tf
+        version = re.compile(r'[0-9]{8}').search(line.split("/")[-1])
```

Comment:
to be more specific, match '(.*-v)([0-9]{9})', in case there are 6 digits somewhere else in the string.  Then group(1) will have the base OS name for later, and group(2) will have the 6 digit version number. (I haven't checked my regex, but should be something like that)

Also, don't call search() in the same line.  save the re.compile() object, and re-use it later.